src/core/regimes.rkt

```diff
@@ -272,21 +266,25 @@
   ;; d for previous index
   ;; This is where the high level bulk of the algorithm is applied
   ;; We get the final splitpoints by applying add-splitpoints as many times as we want
-  (define-values (pa pb pc pd) (initial))
-  (define-values (fa fb fc fd)
-    (let loop ([pa pa] [pb pb] [pc pc] [pd pd])
-      (define-values (na nb nc nd) (add-splitpoint pa pb pc pd))
+  (define-values (pa pb pd) (initial))
+  (define-values (fa fb fd)
+    ; short circuit if there is no other alts to consider
+    (if (> num-candidates 1)
```

Comment:
Does this save any time? Fine either eay

Reply:
Not that I could tell. I added it to clean up the output well trying to understand what's going on.